packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,217 @@
+
+    // Constructor to initialize the Safe and baseModule instances
+    constructor(
+        Safe _safe,
+        LivenessGuard _livenessGuard,
+        uint256 _livenessInterval,
+        uint256 _minOwners,
+        address _fallbackOwner
+    ) {
+        SAFE = _safe;
+        LIVENESS_GUARD = _livenessGuard;
+        LIVENESS_INTERVAL = _livenessInterval;
+        MIN_OWNERS = _minOwners;
+        FALLBACK_OWNER = _fallbackOwner;
+    }
+
+    /// @notice This function can be called by anyone to remove an owner that has not signed a transaction
+    ///         during the liveness interval. If the number of owners drops below the minimum, then the
+    ///         ownership of the Safe is transferred to the fallback owner.
+    function removeOwner(address owner) external {
+        // Check that the owner to remove has not signed a transaction in the last 30 days
+        require(
+            LIVENESS_GUARD.lastLive(owner) < block.timestamp - LIVENESS_INTERVAL,
+            "LivenessModule: owner has signed recently"
+        );
+
+        // Calculate the new threshold
+        address[] memory owners = SAFE.getOwners();
+        uint256 numOwnersAfter = owners.length - 1;
+        if (_isAboveMinOwners(numOwnersAfter)) {
+            // Call the Safe to remove the owner and update the threshold
+            uint256 thresholdAfter = get75PercentThreshold(numOwnersAfter);
+            address prevOwner = _getPrevOwner(owner, owners);
+            _removeOwner({ _prevOwner: prevOwner, _owner: owner, _threshold: thresholdAfter });
+        } else {
+            // The number of owners is dangerously low, so we wish to transfer the ownership of this Safe
+            // to the fallback owner.
+
+            // Remove owners one at a time starting from the last owner.
+            // Since we're removing them in order from last to first, the ordering will remain constant,
+            // and we shouldn't need to query the list of owners again.
+            address prevOwner;
+            for (uint256 i = owners.length - 1; i > 0; i--) {
+                address currentOwner = owners[i];
+                prevOwner = _getPrevOwner(currentOwner, owners);
+
+                // Call the Safe to remove the owner
+                _removeOwner({ _prevOwner: prevOwner, _owner: currentOwner, _threshold: 1 });
+            }
+
+            prevOwner = _getPrevOwner(owners[0], owners);
+            // Add the fallback owner as the sole owner of the Safe
+            _swapToFallbackOwner({ _prevOwner: prevOwner, _oldOwner: owners[0] });
+        }
+
+        _verifyFinalState();
+    }
+
+    /// @notice Sets the fallback owner as the sole owner of the Safe with a threshold of 1
+    /// @param _prevOwner Owner that pointed to the owner to be replaced in the linked list
+    /// @param _oldOwner Owner address to be replaced.
+    function _swapToFallbackOwner(address _prevOwner, address _oldOwner) internal {
```

Comment:
Depends what you mean: [Safe.swapOwner](https://github.com/safe-global/safe-contracts/blob/e870f514ad34cd9654c72174d6d4a839e3c6639f/contracts/base/OwnerManager.sol#L99) itself does not require that `_prevOwner` is the only owner. 

But we're using it because Safe won't allow you to have zero owners (there's no direct check on this, but a combination of threshold and owner count requirements adds up to it). So we can't use `removeOwner` on the last owner, we need to do a swap instead.